### PR TITLE
Add some failure conditions for tracing module

### PR DESCRIPTION
**Pull request by Zhaoningx**
Add some failure conditions for tracing module, and check the
backend to verify the tracing output data is expect.

Signed-off-by: NingX Zhao <ningx.zhao@intel.com>

### CODE REVIEW

File: tests/subsys/tracing/tracing_api/src/main.c [hunk 1]
```diff
@@ -230,6 +245,30 @@ void test_tracing_data_format(void)
...
+
+	/* Disable tracing */
+	tracing_cmd_handle("disable", 8);
+	tracing_format_data(&tracing_error_data, 1);
+	k_sleep(K_MSEC(10));
+	zassert_false(error_data_format_found, "Failded to check output from backend");
+	tracing_format_raw_data(tracing_error_raw_data.data, tracing_error_raw_data.length);
+	k_sleep(K_MSEC(10));
+	zassert_false(error_raw_data_format_found, "Failded to check output from backend");
+	/* Keep default */
```
Comment (enjiamai) [inline tests/subsys/tracing/tracing_api/src/main.c:270]: recommend having a empty line here.

**Review by enjiamai — APPROVED** [review]

File: tests/subsys/tracing/tracing_api/src/main.c [hunk 2]
```diff
@@ -230,6 +245,30 @@ void test_tracing_data_format(void)
...
 	k_sleep(K_MSEC(100));
 	zassert_true(raw_data_format_found == true, "Failded to check output from backend");
+
+	/* make tracing buffer full */
+	uint8_t *buf = data;
+
+	while (tracing_buffer_put_claim(&buf, 5)) {
+		;
+	}
+
```
Comment (LeiW000) [inline tests/subsys/tracing/tracing_api/src/main.c:255]: It is better to change the code like
do {
  ret = tracing_buffer_put_clain();
}while (ret)

**Review by LeiW000 — CHANGES_REQUESTED** [review]

**Comment by Zhaoningx** [comment]
Hi @LeiW000 @enjiamai, I have updated the PR,  could you review it again, thanks.

**Comment by Zhaoningx** [comment]
Hi @enjiamai @nashif @LeiW000 could you help to review this PR? Thanks.

**Comment by enjiamai** [comment]
Hi, @Zhaoningx, please rebase them and check it again, thanks.

**Pull request closed without merging**